### PR TITLE
Add the Eclipse Collections reference guide.

### PR DESCRIPTION
**Pull request by motlin**

### CODE REVIEW

**Comment by motlin** [comment]
I'm not sure why an extra travis build got kicked off but the main builds passed on this PR.

**Comment by motlin** [comment]
The guide document is too big to render in the normal view where we can leave line comments. I've made the suggested correction in a second commit.

File: docs/guide.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ This Guide is an introduction to basic Eclipse Collections concepts and its comm
 []() Getting Started With Eclipse Collections
 =============================================
 
-It is recommended that first time users go through the EC Kata. The Kata is a fun way to help you learn idiomatic Eclipse Collections usage through coding exercises. The Kata can be found on our GitHub page: <https://github.com/goldmansachs/gs-collections-kata>. To run the Kata, please refer to the setup instructions found on page 24 of the [Kata document](http://www.goldmansachs.com/gs-collections/documents/GS%20Collections%20Training%20Session%20and%20Kata%202.0.0.pdf#page=24).
+It is recommended that first time users go through the [Eclipse Collections Kata](https://github.com/eclipse/eclipse-collections-kata). The Kata is a fun way to help you learn idiomatic Eclipse Collections usage through coding exercises. To run the Kata, please refer to the setup instructions found on page 24 of the [Kata document](http://www.goldmansachs.com/gs-collections/documents/GS%20Collections%20Training%20Session%20and%20Kata%202.0.0.pdf#page=24).
```
Comment (itohro) [inline docs/guide.md]: As the last link is still pointing to gs-collections, we could probably just take this out. Set-up instruction is already in the kata README.md anyways. 

> To run the Kata, please refer to the setup instructions found on page 24 of the [Kata document](http://www.goldmansachs.com/gs-collections/documents/GS%20Collections%20Training%20Session%20and%20Kata%202.0.0.pdf#page=24).

Comment (motlin) [inline docs/guide.md]: Done!

**Comment by itohro** [comment]
Thanks! Merged.

**Pull request merged**